### PR TITLE
Make DGContext2D a class instead.

### PR DESCRIPTION
**Pull request by dsteinmo**

### CODE REVIEW

File: src/TriangleNodesProvisioner.cpp [hunk 1]
```diff
@@ -889,6 +868,30 @@ namespace blitzdg {
...
+            Dr.get(),
+            Ds.get(),
+            Lift.get(),
+            vmapM.get(),
+            vmapP.get(),
+            BCmap.get()
+        );
+
+        return *dgContext;
     }
```
Comment (k7miller) [inline src/TriangleNodesProvisioner.cpp:895]: Allocating `dgContext` with new and returning it by const reference has the potential to create a memory leak. You should instead return by value. Don't worry about the overhead of returning by value - DGContext2D is so light-weight it doesn't matter - but even if it, did the return will be either moved or elided away.

Comment (dsteinmo) [reply src/TriangleNodesProvisioner.cpp:895]: Interesting, thanks!

**Review by k7miller — CHANGES_REQUESTED** [review]
Looks good, but please see my comments about returning `dgContext` by const reference.

File: src/TriangleNodesProvisioner.cpp (hunk 1 above)
Comment (k7miller) [reply src/TriangleNodesProvisioner.cpp:895]: I would just write `return DGContext2D{...input params...};` obviously with the input parameters filled-in.

**Pull request merged**